resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v2/InitContainerResourceStagingServerSslOptionsProvider.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.kubernetes.submit.v2
+
+/**
+ * Provides the SSL configurations that should be used when the init-container contacts the
+ * resource staging server. This is responsible for checking if the options for the init-
+ * container should be different from the options used for the submission client.
+ */
+trait InitContainerResourceStagingServerSslOptionsProvider {
+
+  def getSslOptions(): InitContainerResourceStagingServerSslOptions
```

Comment:
I think this was an original idea I had but abandoned it later, forgot to remove the class.